core/src/main/java/org/apache/calcite/sql/fun/SqlLibrary.java

```diff
@@ -94,11 +97,21 @@ public enum SqlLibrary {
   /** Parses a comma-separated string such as "standard,oracle". */
   public static List<SqlLibrary> parse(String libraryNameList) {
     final ImmutableList.Builder<SqlLibrary> list = ImmutableList.builder();
-    for (String libraryName : libraryNameList.split(",")) {
-      SqlLibrary library =
-          requireNonNull(SqlLibrary.of(libraryName),
-              () -> "library does not exist: " + libraryName);
-      list.add(library);
+    List<String> libList = Arrays.asList(libraryNameList.split(","));
```

Comment:
Does this do the right thing if someone writes 'all,spatial'? I suspect not, but even if it does, it's not clear.

To simplify, change the method contract so that it may contain duplicates, may contain 'all', and may not be sorted. Then deal with expanding 'all' in the code that calls the method.

Reply:
I have changed this function, please have another look when you have time.

Reply:
Hi @julianhyde, may I ask if you have time to review this PR, thanks very much